test/extended/apiserver/resiliency.go

```diff
@@ -0,0 +1,121 @@
+package apiserver
+
+import (
+	"context"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"time"
+
+	"github.com/onsi/ginkgo"
+	"github.com/onsi/gomega"
+	configv1 "github.com/openshift/api/config/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	types "k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/util/uuid"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/client-go/rest"
+	"k8s.io/kubernetes/test/e2e/framework"
+	e2eskipper "k8s.io/kubernetes/test/e2e/framework/skipper"
+
+	"github.com/openshift/origin/test/extended/scheme"
+	"github.com/openshift/origin/test/extended/single_node"
+	exutil "github.com/openshift/origin/test/extended/util"
+)
+
+var _ = ginkgo.Describe("[Conformance][sig-sno][Serial] Cluster", func() {
+	f := framework.NewDefaultFramework("cluster-resiliency")
+	f.SkipNamespaceCreation = true
+	f.SkipPrivilegedPSPBinding = true
+
+	oc := exutil.NewCLIWithoutNamespace("cluster-resiliency")
+
+	ginkgo.It("should allow a fast rollout of kube-apiserver", func() {
+		controlPlaneTopology, _ := single_node.GetTopologies(f)
+
+		if controlPlaneTopology != configv1.SingleReplicaTopologyMode {
+			e2eskipper.Skipf("Test is only relevant for single replica topologies")
+		}
+
+		config, err := framework.LoadConfig()
+		framework.ExpectNoError(err)
+
+		setRESTConfigDefaults(config)
+		restClient, err := rest.RESTClientFor(config)
+		framework.ExpectNoError(err)
+
+		httpClient := restClient.Client
+
+		ginkgo.By("Making sure no previous rollout is in progress")
+		clusterApiServer, err := oc.AdminOperatorClient().OperatorV1().KubeAPIServers().Get(context.Background(), "cluster", metav1.GetOptions{})
+		framework.ExpectNoError(err)
+		gomega.Expect(clusterApiServer.Status.NodeStatuses[0].TargetRevision).To(gomega.Equal(int32(0)))
+
+		ginkgo.By("Forcing API rollout")
+		forceApiRollout(oc)
+
+		ginkgo.By("Expecting API to become unavailable")
+		wait.PollImmediate(time.Second, time.Minute, func() (bool, error) {
```

Comment:
what if it is not unavaiable within 1 minute? woudln't that skew the next test?

Reply:
That's possible
Do we have some way to stop the whole suite from running?